smart-home-2020-master/src/main/java/ru/sbt/mipt/oop/Application.java

```diff
@@ -1,35 +1,33 @@
 package ru.sbt.mipt.oop;
 
-import com.google.gson.Gson;
+import java.util.Arrays;
+import java.util.List;
 
-import java.io.IOException;
-import java.nio.file.Files;
-import java.nio.file.Paths;
+public class Application {
+    private final EventCreator eventCreator;
+    private final SmartHome smartHome;
 
-import static ru.sbt.mipt.oop.SensorEventType.*;
 
-public class Application {
+    public Application(EventCreator eventCreator, SmartHome smartHome) {
+        this.eventCreator = eventCreator;
+        this.smartHome = smartHome;
+    }
 
-    public static void main(String... args) throws IOException {
+    public void run(){
+        List<EventProcessor> processEvents = Arrays.asList(new EventProcessorDecorator(new LightEventProcessor()), new EventProcessorDecorator(new DoorsEventProcessor()), new EventProcessorDecorator(new HallDoorEventProcessor()), new EventProcessorDecorator(new SignalingEventProcessor()));
```

Comment:
https://github.com/iagoshkov/smart-home-2020/blob/master/STUDENT_GUIDE.md#%D0%BF%D0%B5%D1%80%D0%B5%D0%B4%D0%B0%D0%B2%D0%B0%D0%B9%D1%82%D0%B5-%D0%B7%D0%B0%D0%B2%D0%B8%D1%81%D0%B8%D0%BC%D0%BE%D1%81%D1%82%D0%B8-%D1%81%D0%BD%D0%B0%D1%80%D1%83%D0%B6%D0%B8